gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -1306,6 +1307,83 @@ gum_do_enumerate_exports (GumEnumerateExportsContext * ctx,
   return carry_on;
 }
 
+kern_return_t
+gum_malloc_memory_reader (task_t remote_task,
+    vm_address_t remote_address, vm_size_t size, void ** local_memory) {
+
+  *local_memory = (void *) remote_address;
+
+  return KERN_SUCCESS;
+}
+
+struct _gum_malloc_helper_info {
+  GumFoundMallocRangeFunc func;
+  gpointer user_data;
+  gboolean should_abort;
+};
+
+void
+gum_process_enumerate_malloc_ranges_helper (task_t task,
+    void * user_data, unsigned type,
+    vm_range_t * ranges, unsigned count) {
+
+  struct _gum_malloc_helper_info * info =
+    (struct _gum_malloc_helper_info *) user_data;
+
+  if (info->should_abort) {
+    return;
+  }
+
+  GumMemoryRange gum_range;
+  GumMallocRangeDetails details;
+  details.range = &gum_range;
+
+  for (unsigned i = 0; i != count; i++) {
+    vm_range_t range = ranges[i];
+
+    gum_range.base_address = range.address;
+    gum_range.size = range.size;
+
+    gboolean ret = info->func (&details, info->user_data);
+    if (!ret) {
+      info->should_abort = TRUE;
+      return;
+    }
+  }
+}
+
+void gum_process_enumerate_malloc_ranges (GumFoundMallocRangeFunc func,
+    gpointer user_data) {
+
+  task_t task = mach_task_self ();
+
+  vm_address_t *malloc_zone_addresses;
```

Comment:
Another unwritten style rule: always space after the type. So here it should be:
```C
vm_address_t * malloc_zone_addresses;
```
(and the declaration should be moved up to the beginning of the scope)